src/SpaceTime/undirected/Unit.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using Bearded.Utilities.Math;
+using OpenTK;
+
+namespace Bearded.Utilities.SpaceTime
+{
+    /// <summary>
+    /// A type-safe representation of an undirected signed distance or length.
+    /// </summary>
+    public struct Unit : IBackedBy<float>, IEquatable<Unit>, IComparable<Unit>
+    {
+        private readonly float value;
+
+        #region construction
+
+        /// <summary>
+        /// Creates a new instance of the Unit type.
+        /// </summary>
+        public Unit(float value)
+        {
+            this.value = value;
+        }
+
+        #endregion
+
+        #region properties
+
+        /// <summary>
+        /// Returns the numeric value of the unit value.
+        /// </summary>
+        public float NumericValue { get { return this.value; } }
+
+        /// <summary>
+        /// Returns the type-safe square of the unit value.
+        /// </summary>
+        public Squared<Unit> Squared { get { return Squared<Unit>.FromRoot(this.value); } }
+
+        /// <summary>
+        /// Returns a Unit type with value 0.
+        /// </summary>
+        public static Unit Zero { get { return new Unit(0); } }
+        /// <summary>
+        /// Returns a Unit type with value 1.
+        /// </summary>
+        public static Unit One { get { return new Unit(1); } }
+
```

Comment:
You mean, you prefer `Unit.OfValue(3)` over `new Unit(3)`? (Also note that since these are value types, we cannot make the default constructor private, so instantiation with `new` will always be possible - if this is what you are concerned about.)